source/adios2/core/Attribute.h

```diff
@@ -13,11 +13,23 @@
 
 #include "adios2/core/AttributeBase.h"
 
+#include <type_traits>
+
 namespace adios2
 {
 namespace core
 {
 
+template <class T>
+struct requires_zero_padding : std::false_type
+{
+};
+
+template <>
+struct requires_zero_padding<long double> : std::true_type
+{
```

Comment:
I feel like this is a pretty unique case only used by Attribute to address a specific issue.  As such it would be more appropriate to move to the `.cpp` file as an internal helper rather than a public type.  Also, just for convention with the rest of the code base, `requires_zero_padding` should be changed to `RequiresZeroPadding`.

Reply:
The idea behind putting it in the header was that if there are any non-primitive types that get used in `Attribute` in the future, they can declare `RequiresZeroPadding` in their own header. If you don't foresee that happening then I'll go ahead and move it.

Reply:
> The idea behind putting it in the header was that if there are any non-primitive types that get used in Attribute in the future, they can declare RequiresZeroPadding in their own header.  That'd work if it was propagated to the bindings but this is in the `core` which isn't user-facing.  It's the back end implementation for the user-facing `bindings`.  I really think this is specific to the case of `long double`.  There's *lots* of other internal structre that needs to be addressed if we were to support user-defined types, if ever, so I think it's fine to just move it internal.